backend/src/httpserver/myapp/views.py

```diff
@@ -195,40 +202,161 @@ def calculate_midpoint(lat_longs):
     central_square_root = math.sqrt(x * x + y * y)
     central_latitude = math.atan2(z, central_square_root)
 
-    return (math.degrees(central_latitude), math.degrees(central_longitude))
+    return math.degrees(central_latitude), math.degrees(central_longitude)
 
 
 def get_friends(request):
     """
     Handles the /friends endpoint.
     """
-    if request.method == 'GET':
-        data = request.GET
-    return HttpResponse('Thanks for submitting your data')
+    try:
+        if request.method == 'GET':
+            data = request.GET
+            if 'HTTP_AUTHORIZATION' not in request.META:
+                return JsonResponse({'error': 'access forbidden'}, status=401)
+            authentication = request.META['HTTP_AUTHORIZATION'][7:]  # starts with "Bearer "
+            email = data['email']
+            bearer_manager = Bearer.objects
+            if bearer_manager.verify_token(authentication, email):
+                db = BitdotioDB(os.getenv(token))
+                # Authentication bearer should guarantee that user exists
+                friend_list = db.db_query(f'SELECT {DB_FRIENDS}.user1, {DB_USERS}.address, '
```

Comment:
Currently seems like we have some SQL injection vulnerabilities. Probably want to fix this in the future